Activities/Scripting/UiPath.Scripting.Activities/PowerShell/ExecutePowerShellCore.cs

```diff
@@ -0,0 +1,396 @@
+                }
+                pipeline.Input.Close();
+
+            }
+            catch
+            {
+                if (runspace != null)
+                {
+                    runspace.Dispose();
+                }
+
+                if (pipeline != null)
+                {
+                    pipeline.Dispose();
+                }
+                if (!ContinueOnError.Get(context))
+                {
+                    throw;
+                }
+            }
+
+            context.UserState = pipeline;
+            return new PipelineInvokerAsyncResult(pipeline, callback, state);
+        }
+
+        // Called by the runtime after execution of this asynchronous activity.
+        protected override void EndExecute(AsyncCodeActivityContext context, IAsyncResult result)
+        {
+            PipelineInvokerAsyncResult asyncResult = result as PipelineInvokerAsyncResult;
+            Pipeline pipeline = context.UserState as Pipeline;
+
+            if (asyncResult == null || pipeline == null) return;
+            try
+            {
+                if (asyncResult.Exception != null)
+                {
+                    throw asyncResult.Exception;
+                }
+                if (asyncResult.ErrorRecords != null && asyncResult.ErrorRecords.Count > 0)
+                {
+                    if (asyncResult.ErrorRecords.Count == 1)
+                    {
+                        throw asyncResult.ErrorRecords[0].Exception;
+                    }
+
+                    string exceptionMessage = Resources.PowerShellExceptionMessage;
+                    foreach (var error in asyncResult.ErrorRecords)
+                    {
+                        exceptionMessage += System.Environment.NewLine + error.Exception.Message;
+                    }
+
+                    throw new Exception(exceptionMessage);
+                }
+                this.PipelineOutput.Set(context, asyncResult.PipelineOutput);
+                this.Errors.Set(context, asyncResult.ErrorRecords);
+
+                foreach (KeyValuePair<string, Argument> entry in this.PowerShellVariables)
+                {
+                    if ((entry.Value.Direction == ArgumentDirection.Out) || (entry.Value.Direction == ArgumentDirection.InOut))
+                    {
+                        object value = pipeline.Runspace.SessionStateProxy.GetVariable(entry.Key);
+                        entry.Value.Set(context, value);
+                    }
+                }
+            }
+            catch
+            {
+                if (!ContinueOnError.Get(context)) throw;
+            }
+            finally
+            {
+                DisposePipeline(pipeline);
+            }
+        }
+    }
+
+    // An instance of this object is returned to the runtime upon beginning
+    // execution.  This also keeps a pointer to a callback function, which
+    // it calls when it completes.
+    internal class PipelineInvokerAsyncResult : IAsyncResult
```

Comment:
Please move this class in a different file, so that the code will be more organized.